src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -150,7 +150,9 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
             File file = imageIdFile.get().asFile
             if(file.exists()) {
                 try {
-                    getDockerClient().inspectImageCmd(file.text).exec()
+                    def fileImageId = file.text
+                    getDockerClient().inspectImageCmd(fileImageId).exec()
+                    imageId.set(fileImageId)
```

Comment:
The value of the `imageId` property should be set before making the call to Docker. In case Docker communication throws an exception, we already have the field populated.

Reply:
Interesting, okay I performed the set afterwards in case the inspectImageCmd had thrown due to the image id not being found, setting it to an invalid value would have seemed incorrect in that use case. 

I hadn't thought about the use case that the communication to docker itself had failed. 

Is it worth making the distinction? I.E. set if the failure is due to a communication issue and assuming it's still a valid image id and not setting it if the image doesn't exist. A quick look through the docker-java seems to mention a specific NotFoundException (would need to validate this)

Reply:
@bmuschko : I think we should set the imageIdt after the inspect fails because when it fails the task is set as not up-to-date and the imageId should not be set. It should be set only when the inspect succeeds in my opinion.
I will add the check in the tests though it is a good point.

@stevendpclark : At first I was catching only NotFoundException but then I realized that I could get more in other situations, miscommunication being one of them.

Reply:
@llaforest I think you are right here. Sorry, let's change this back.